src/deploymentSlotActions.ts

```diff
@@ -0,0 +1,99 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AzureAccountWrapper } from './azureAccountWrapper';
+import { WizardBase, WizardResult, WizardStep, UserCancelledError, QuickPickItemWithData } from './wizard';
+import { SubscriptionModels, ResourceManagementClient, ResourceModels } from 'azure-arm-resource';
+import WebSiteManagementClient = require('azure-arm-website');
+import models = require('azure-arm-website');
+import { NodeBase } from './nodeBase';
+import { DeploymentSlotNode } from './deploymentSlotNodes';
+import * as WebSiteModels from '../node_modules/azure-arm-website/lib/models';
+import * as util from './util';
+
+export class DeploymentSlotSwapper extends WizardBase {
+    constructor(output: vscode.OutputChannel, readonly azureAccount: AzureAccountWrapper, readonly slot: DeploymentSlotNode) {
+        super(output);
+        this.steps.push(new SwapStep(this, azureAccount,slot));
+    }
+
+    protected onExecuteError(step: WizardStep, stepIndex: number, error: Error) {
+        if (error instanceof UserCancelledError) {
+            return;
+        }
+        this.writeline(`Failed to swap deployment slots - ${error.message}`);
+    }
+}
+
+class SwapStep extends WizardStep {
+    private _subscription: SubscriptionModels.Subscription;
+    private _sourceSlot;
+    private _targetSlot;
+    
+    get sourceSlot(): DeploymentSlotNode {
+        return this._sourceSlot;
+    }
+    get targetSlot(): DeploymentSlotNode {
+        return this._targetSlot;
+    }
+    set sourceSlot(slot: DeploymentSlotNode) {
+        this._sourceSlot = slot;
+    }
+    set targetSlot(slot: DeploymentSlotNode) {
+        this._targetSlot = slot;
+    }
+
+    constructor(wizard: WizardBase, readonly azureAccount: AzureAccountWrapper, readonly slot: DeploymentSlotNode) {
+        super(wizard, 'Select a slot to swap with');
+        this.sourceSlot = slot;
+    }
+
+    async prompt(): Promise<void> {
+        const deploymentSlots: DeploymentSlotNode[] = await this.slot.parent.getChildren(this.azureAccount);
+        let otherSlots: QuickPickItemWithData<null>[] = [];
+
+        for (let slot of deploymentSlots) {
+            if (this.slot.label !== slot.label) {
+                // Deployment slots must have an unique name
+                const otherSlot: QuickPickItemWithData<null> = {

```

Comment:
I think the idea of QuickPickItemWithData is to set `slot` as the data. That way you can just use result.data instead of doing the 'find' on line 75